crates/common/src/fs.rs

```diff
@@ -67,6 +79,20 @@ pub fn write_pretty_json_file<T: Serialize>(path: &Path, obj: &T) -> Result<()>
     writer.flush().map_err(|e| FsPathError::write(e, path))
 }
 
+/// Writes the object as a gzip compressed file.
+pub fn write_json_gzip_file<T: Serialize>(path: &Path, obj: &T) -> Result<()> {
+    let file = create_file(path)?;
+    let writer = BufWriter::new(file);
+    let mut encoder = GzEncoder::new(writer, Compression::default());
+    serde_json::to_writer(&mut encoder, obj)
+        .map_err(|source| FsPathError::WriteJson { source, path: path.into() })?;
```

Comment:
nit: this is a serialization error rather than an I/O error

Reply:
could you pls suggest how to update, is that a different `FsPathError` to add?